src/export/StructureDefinitionExporter.ts

```diff
@@ -155,18 +310,119 @@ export class StructureDefinitionExporter implements Fishable {
+    const pathType = extractPathTypeFromStructDefType(structDef.type);
+    elements.forEach(e => {
+      e.id = e.id.replace(/^[^.]+/, pathType);
+    });
+
+    // The root element's base.path should be the same as root element's path
+    elements[0].base.path = elements[0].path;
+
+    if (
+      fshDefinition.parent === 'Element' ||
+      fshDefinition.parent === 'http://hl7.org/fhir/StructureDefinition/Element'
+    ) {
+      // The FHIR Element StructureDefinition contains extensions related to being
+      // defined as 'normative'. These extensions should not be carried forward in
+      // the new StructureDefinition.
+      delete elements[0].extension;
+    }
+
+    structDef.elements = elements;
+    structDef.captureOriginalElements();
+
+    // The following changes to the root element will be included in the
+    // differential root element.
+
+    // Reset the root element's short and definition
+    if (fshDefinition.title) {
+      structDef.elements[0].short = fshDefinition.title;
+    }
+    if (fshDefinition.description) {
+      structDef.elements[0].definition = fshDefinition.description;
     }
   }
 
   /**
    * Sets the rules for the StructureDefinition
    * @param {StructureDefinition} structDef - The StructureDefinition to set rules on
    * @param {Profile | Extension} fshDefinition - The Profile or Extension we are exporting
+   * @private
    */
   private setRules(structDef: StructureDefinition, fshDefinition: Profile | Extension): void {
     resolveSoftIndexing(fshDefinition.rules);
     for (const rule of fshDefinition.rules) {
+      // Specific rules are permitted for each structure definition type
+      // (i.e., Profile, Logical, etc.). Log an error for disallowed rules
+      // and continue to next rule.
+      if (!isAllowedRule(fshDefinition, rule)) {
+        logger.error(
+          `Use of '${rule.constructorName}' is not permitted for '${fshDefinition.constructorName}'. Skipping '${rule.constructorName}' at path '${rule.path}' for '${fshDefinition.name}'.`,
+          rule.sourceInfo
+        );
+        continue;
+      }
+
+      if (rule instanceof AddElementRule) {
+        try {
+          // Note: newElement() method automatically adds the new element to its structDef.elements
+          const newElement = structDef.newElement(rule.path);
+          newElement.applyAddElementRule(rule, this);
+        } catch (e) {
+          logger.error(e.message, rule.sourceInfo);
+        }
+        continue;
+      }
+
       const element = structDef.findElementByPath(rule.path, this);
+
+      if (element && (fshDefinition instanceof Logical || fshDefinition instanceof Resource)) {
+        // The FHIR spec prohibits constraining any parent element in a 'specialization'
+        // (i.e., logical model and resource), therefore log an error if that is attempted
+        // and continue to the next rule.
+        if (element.path !== element.base.path) {
```

Comment:
I'm noticing that this check prevents something like this:
```
Logical: MyLogicalBase
Parent: Base
* foo 1..1 HumanName "short"
* foo.given 1..1
```
which may be too aggressive? I'm not entirely sure. Someone could work around this by profiling HumanName and referring to that type instead, so it isn't impossible to represent this idea, but maybe more difficult. What do you think @joepaquette, should this be allowed?

Reply:
Great question! I think this is probably a reasonable expectation so I think I should fix this. @cmoesel ??

Reply:
@cmoesel @ngfreiter  Another question... What constitutes a parent? In the example above, we could also try `* foo.id 1..1`. In this case, I don't think this should be allowed since `Element` is the parent of `HumanName`. Does this make sense?

Reply:
I should look back at the part of the spec where this is discussed, but my interpretation was that you cannot constrain any path that would be a valid path in the parent SD.  So, since `foo` was only introduced in `MyLogicalBase`, then I think you could constrain _any_ path that starts with `foo` inside `MyLogicalBase` -- because there was no `foo` anything in the parent.

Reply:
I'm trying to find where we got the idea that you cannot constrain elements from the parent.  And having trouble finding it (although I _know_ we discussed this and had justification for it).  I wouldn't make any code changes about this until we chase it down a little further.

Reply:
[Found it] and asked a follow-up question.